packages/optimizely-sdk/lib/optimizely/index.ts

```diff
@@ -1415,4 +1425,159 @@ export default class Optimizely {
 
     return Promise.race([this.readyPromise, timeoutPromise]);
   }
+
+  //============ decide ============//
+
+  /**
+   * Creates a context of the user for which decision APIs will be called.
+   *
+   * A user context will be created successfully even when the SDK is not fully configured yet.
+   *
+   * @param  {string}          userId      The user ID to be used for bucketing.
+   * @param  {UserAttributes}  attributes  Optional user attributes.
+   * @return {OptimizelyUserContext|null}  An OptimizelyUserContext associated with this OptimizelyClient or
+   *                                       null if client instance or provided inputs are invalid
+   */
+  createUserContext(userId: string, attributes?: UserAttributes): OptimizelyUserContext | null {
+    if (!this.validateInputs({ user_id: userId }, attributes)) {
+      return null;
+    }
+
+    return new OptimizelyUserContext({
+      optimizely: this,
+      userId,
+      attributes
+    });
+  }
+
+  /**
+   *
+   * Get all decide options.
+   *
+   * @param  {OptimizelyDecideOptions[]}          options     Optional decide options
+   * @return {OptimizelyDecideOptions[]}          Array of all provided decide options
+   */
+  private getAllDecideOptions(options?: OptimizelyDecideOptions[]): OptimizelyDecideOptions[] {
+    const allDecideOptions = this.defaultDecideOptions ?? [];
```

Comment:
We don't want to mutate `this.defaultDecideOptions`:
```suggestion
    const allDecideOptions = [...this.defaultDecideOptions] ?? [];
```
Also, could remove the last part if we ensure `this.defaultDecideOptions` is an array.